Update role-mapping-resources.asciidoc (#110420)
elastic/elasticsearch: docs/reference/rest-api/security/role-mapping-resources.asciidoc

`groups`::
(array of strings) The groups to which the user belongs. For example, `"groups" : [ "cn=admin,ou=groups,dc=example,dc=com","cn=esusers,ou=groups,dc=example,dc=com ]`.
`metadata`::

> (object) Additional metadata for the user. This can include a variety of key-value pairs. When referencing metadata fields in role mapping rules, use the dot notation to specify the key within the metadata object. If the key contains special characters such as parentheses, dots, or spaces, you must escape these characters using backslashes (`\`). For example, `"metadata": { "cn": "John Smith" }`.

Reviewer: where "for example" is placed seems to indicate it's an example for escaped chars. consider:

```suggestion
(object) Additional metadata for the user, passed as key-value pairs. For example, `"metadata": { "cn": "John Smith" }`.
+
When referencing metadata fields in role mapping rules, use dot notation to specify the key within the metadata object. If the key contains special characters such as parentheses, dots, or spaces, you must escape these characters using backslashes (`\`). 
```

also, `the dot notation` should just be `dot notation`.

consider providing a second example with multiple KV pairs, dot notation, and escaped characters.
